doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+* ``tree.Criterion``
+
+.. _glossary_target_types:
+
+Target Types
+============
+
+.. glossary::
+
+    binary
+        A classification problem consisting of two classes.  A binary target
+        may represented as for a :term:`multiclass` problem but with only two
+        labels.  A binary decision function is represented as a 1d array.
+
+        Note that a dataset sampled from a multiclass ``y`` or a continuous
+        ``y`` may appear to be binary.
+
+        :func:`~utils.multiclass.type_of_target` will return 'binary' for
+        binary input, or a similar array with only a single class present.
+
+    continuous
+        TODO
+
+    multiclass
+        A classification problem consisting of more than two classes.  A
+        multiclass target may be represented as a 1-dimensional array of
+        strings or integers. A 2d column vector of integers (i.e. a
+        single output is also accepted.
+
+        We may also support other orderable, hashable objects as class labels.
+
+        For semi-supervised classification, :term:`unlabeled` samples should
+        have the special label -1 in ``y``.
+
+        Within sckit-learn, all estimators supporting binary classification
+        also support multiclass classification, using One-vs-Rest by default.
+
+        A :class:`preprocessing.LabelEncoder` helps to canonicalize multiclass
+        targets as integers.
+
+        :func:`~utils.multiclass.type_of_target` will return 'multiclass' (or
+        'binary' in the degenerate case) for multiclass input.
+
+    multilabel
+        A multioutput target where each output is :term:`binary`.  This may be
+        represented as a 2d (dense) array or sparse matrix of integers, such
+        that each column is a separate binary target, where positive labels are
+        indicated with 1 and negative labels are usually -1 or 0.
+
+        Semantically, a multilabel target can be thought of as a set of labels
+        for each sample.  While not used internally,
+        :class:`preprocessing.MultiLabelBinarizer` is provided as a utility to
+        convert from a list of sets representation to a 2d array or sparse
+        matrix.
+
+        :func:`~utils.multiclass.type_of_target` will return
+        'multilabel-indicator' for multilabel input, whether sparse or dense.
+
+    multioutput continuous
+        TODO
+
+    multioutput multiclass
+        A classification problem .
+
+        TODO
+
+Methods
+=======
+
+.. glossary::
+
+    ``decision_function``
+
+        In a fitted :term:`classifier` or :term:`outlier detector`, predicts a
+        "soft" score for each sample in relation to each class, rather than the
+        "hard" categorical prediction produced by :term:`predict`.
+
+        Output conventions:
+
+        binary classification
+            A 1-dimensional array, where values strictly greater than zero
+            indicate the positive class (i.e. the last class in
+            :term:`classes_`).
```

Comment:
ah, here it is ;) I would also mention that above or link to this place?

Reply:
Do we require ``classes_`` to be ordered?

Reply:
covered elsewhere...